### PR TITLE
Bookerang implemented + projectile prefab

### PR DESCRIPTION
**Pull request by Thorvers**
Throw book, it returns

### CODE REVIEW

File: RPG Wizardry/Assets/Scripts/Avatar/AvatarManager.cs [hunk 1]
```diff
@@ -13,6 +13,11 @@ public class AvatarManager : SingletonBehaviour<AvatarManager>, IHealth
         /// Player Health
         /// </summary>
         public ushort Health { get; private set; }
+
+        /// <summary>
+        /// Renderer of the "crosshair" book, necessary for bookerang spell
+        /// </summary>
+        public SpriteRenderer BookRenderer;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Avatar/AvatarManager.cs]: This shouldn't be public set

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs [hunk 2]
```diff
@@ -1,11 +1,147 @@
...
+
+namespace nl.SWEG.RPGWizardry.Sorcery.Spells
 {
     public class Bookerang : Projectile
     {
+        #region Variables
+        /// <summary>
+        /// Bool for whether the book is moving forwards (false) or returning (true)
+        /// </summary>
+        private bool back = false;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: No need to set default values

**Review by Sanko-Kallig — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs [hunk 3]
```diff
@@ -1,11 +1,147 @@
...
+        /// <summary>
+        /// Move in a straight line until we hit something or max out our range
+        /// Wait for a short time, then return to the player
+        /// </summary>
         protected override void Move()
         {
-            //turn around???
-            base.Move();
+            //SPIN TO WIN
+            spriteTransform.Rotate(Vector3.forward,1500 * Time.deltaTime,Space.World);
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: Move rotationspeed to a (editor-)variable

**Review by frankvHoof93 — CHANGES_REQUESTED** [review]

**Review by Sven-dev — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs (hunk 2 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: Force of habit. Fixed.

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs (hunk 3 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: Sure

File: RPG Wizardry/Assets/Scripts/Avatar/AvatarManager.cs (hunk 1 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Avatar/AvatarManager.cs]: Fixed

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs [hunk 4]
```diff
@@ -1,11 +1,154 @@
...
+namespace nl.SWEG.RPGWizardry.Sorcery.Spells
 {
     public class Bookerang : Projectile
     {
+        #region Variables
+        #region Public
+        /// <summary>
+        /// Speed at which the book rotates in flight
+        /// </summary>
+        public int SpinSpeed;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: Does this need a public set?

**Review by frankvHoof93 — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs (hunk 4 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Sorcery/Spells/Bookerang.cs]: Dangit

**Pull request merged**